### ISSUE TITLE
Group of global options lost on child command

### ISSUE THREAD

**Issue by raijinsetsu** [yargs/yargs#1081]
I found that the groups do not propagate down to child commands, though the options do. Sample code:

```typescript
let yargs = Yargs
    .strict()
    .help('h')
    .alias('h', 'help')
    .option('logFacility', {
        description: 'sets the log facility',
        default: process.env.LOGFACILITY || 'default',
        choices: ['console', 'file', 'null', 'default']
    })
    .option('logFile', {
        description: 'the log file (without extension) to be used by the "file" log facility',
        string: true,
        default: process.env.LOGFILE || path.join('logs', 'command')
    })
    .option('logRotate', {
        description: 'controls how and when the log file is rotated',
        string: true,
        default: process.env.LOGROTATE || '1D:2:5'
    })
    .option('logLevel', {
        description: 'controls the level of logging; the default is taken from the LOGLEVEL environment variable if possible, or defaults to "info"',
        default: process.env.LOGLEVEL || 'info',
        choices: ['none', 'error', 'warning', 'info', 'debug']
    })
    .group(['logFacility', 'logFile', 'logLevel', 'logRotate'], 'Logging')
    .global(['development', 'logFacility', 'logFile', 'logLevel', 'logRotate'])
    .strict()
    .commandDir('commands');
```

If I execute "node index --help", I see my those logging options in the "Logging" group.
If I execute "node index sub-command --help", they are no longer grouped but the options marked as global appear mixed in with the sub-commands options.

The only workaround I have currently is to specify the "group" at every command level.

**Comment by bcoe** [comment]
@raijinsetsu my expectation as well is that `.group` would be inherited by commands.

**Comment by cspotcode** [comment]
Related to #1173.

**Comment by mleguen** [comment]
Groups are indeed lost, but only after 2 layers of commands, as stated in #1173.

**Comment by bcoe** [comment]
@raijinsetsu this should have been addressed in `yargs@14.2.0`.
